### PR TITLE
[CRED-817][CRED-818] Add list and crud method. Add tests for list method

### PR DESCRIPTION
**Pull request by alanwernicksf**

### CODE REVIEW

File: lib/billdotcom_integration/action.rb [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
-    LOGIN = 'login'
+    LOGIN = 'Login'
+    LIST = 'List'
+
+    module Crud
+      CREATE = 'Create'
+      UPDATE = 'Update'
+      READ = 'Read'
+      DELETE = 'Delete'
+      UNDELETE = 'Undelete'
```
Comment (jibrankalia-sf) [inline lib/billdotcom_integration/action.rb]: crudu?

Comment (jibrankalia-sf) [reply lib/billdotcom_integration/action.rb]: :)

File: lib/billdotcom_integration/rest_client.rb [hunk 2]
```diff
@@ -47,5 +48,24 @@ def login
...
+
+    def intialize_crud_methods!
+      Action::Crud.constants(false).each do |k|
+        v = Object.const_get("BillDotComIntegration::Action::Crud::#{k}")
+        instance_eval <<-CRUD_ACTION
+          def #{v.downcase}(entity, params)
+            api_request = EntityCrudApiRequest.new(entity, v, params)
+            post(api_request)
+          end
+        CRUD_ACTION
```
Comment (jibrankalia-sf) [inline lib/billdotcom_integration/rest_client.rb]: cool stuff

File: lib/billdotcom_integration/action.rb (hunk 1 above)
Comment (sharon-white-sf) [reply lib/billdotcom_integration/action.rb]: lol

**Pull request merged**